sycl/doc/KernelProgramCache.md

```diff
@@ -70,40 +70,83 @@ In order to eliminate this waste of run-time we introduce a kernel and program
+ which has been already loaded and built for target device.
+ - on-disk cache which is used to store device binaries between application
+ executions.
 
+### In-memory cache
 The cache stores underlying PI objects behind `cl::sycl::program` and
 `cl::sycl::kernel` user-level objects in a per-context data storage. The storage
 consists of two maps: one is for programs and the other is for kernels.
 
-The programs map's key consists of three components: kernel set id<sup>[1](#what-is-ksid)</sup>,
-specialized constants, device this program is built for.
+The programs map's key consists of four components: kernel set id<sup>[1](#what-is-ksid)</sup>,
+specialized constants, device this program is built for, build options id <sup>[2](#what-is-bopts)</sup>.
 
-The krnels map's key consists of three components too: program the kernel
-belongs to, kernel name<sup>[2](#what-is-kname)</sup>, device the program is
-built for.
+The kernels map's key consists of four components too: program the kernel
+belongs to, kernel name<sup>[3](#what-is-kname)</sup>, device the program is
+built for, build options id<sup>[2](#what-is-bopts)</sup>.
 
 <a name="what-is-ksid">1</a>: Kernel set id is an ordinal number of the device
 binary image the kernel is contained in.
-
-<a name="what-is-kname">2</a>: Kernel name is a kernel ID mangled class' name
+<a name="what-is-bopts">2</a>: Hash for the string representation of build
+options set in application or environment variables (e.g.
+SYCL_PROGRAM_COMPILE_OPTIONS, SYCL_PROGRAM_LINK_OPTIONS)
+<a name="what-is-kname">3</a>: Kernel name is a kernel ID mangled class' name
 which is provided to methods of `cl::sycl::handler` (e.g. `parallel_for` or
 `single_task`).
 
+### On-disk cache
+The cache is hidden behind in-memory cache and stores the same underlying PI
+object behind `cl::sycl::program` user-level objects in a per-context data
+storage.
+The storage consists of the map for storing programs. It uses different keys to
+address difference in SYCL objects ids between applications runs as well as the
+fact that the same kernel name can be used in different SYCL applications.
+
+The programs map's key consists of four components: device image id<sup>[1](#what-is-diid)</sup>,
+specialized constants, device id<sup>[2](#what-is-did)</sup> this program is
+built for, build options id<sup>[3](#what-is-bopts)</sup>.
+<a name="what-is-diid">1</a>: Hash out of  first 10 kB (for performance reasons) of SPIRV image (?and/or source code?) used as input for the build.
+<a name="what-is-did">2</a>: Hash out of the string which is concatenation of values for
+`info::platform::name`, `info::device::name`, `info::device::version`,
+`info::device::driver_version` parameters to differentiate different HW and SW
+installed on the same host as well as SW/HW upgrades.
+
+## Environment variables
+The following environment variables affect cache mechanism:
+`DPCPP_CACHE_DIR`=/path/to/cache/location
```

Comment:
Is there a default value for the case when the variable isn't set but `DPCPP_CACHE_ENABLED` is set?

Reply:
it planned to be. I have added to the doc.